### PR TITLE
Exclude profiles with labels from history.

### PR DESCRIPTION
**Pull request by blkt**
# Summary

History list showed evaluation results even for profiles having labels. This is different from the behaviour of Profile list and status, which only showed results for profiles without labels.

This change makes the behaviour of History list uniform with Profile list and status.

Fixes stacklok/minder-stories#102

## Change Type

- [X] Bug fix (resolves an issue without affecting existing features)
- [X] Feature (adds new functionality without breaking changes)
- [ ] Breaking change (may impact existing functionalities or require documentation updates)
- [ ] Documentation (updates or additions to documentation)
- [ ] Refactoring or test improvements (no bug fixes or new functionality)

# Testing

Unit tested.

# Review Checklist:

- [X] Reviewed my own code for quality and clarity.
- [X] Added comments to complex or tricky code sections.
- [ ] Updated any affected documentation.
- [ ] Included tests that validate the fix or feature.
- [X] Checked that related changes are merged.

### CODE REVIEW

File: database/query/eval_history.sql [hunk 1]
```diff
@@ -158,6 +158,8 @@ SELECT s.id::uuid AS evaluation_id,
    AND (sqlc.narg(tots)::timestamp without time zone IS NULL OR  s.evaluation_time < sqlc.narg(tots))
    -- implicit filter by project id
    AND j.id = sqlc.arg(projectId)
+   -- implicit filter, exclude all labelled profiles
+   AND p.labels = ARRAY[]::TEXT[]
```
[the remark this answers is not in the document]
Comment (blkt) [reply database/query/eval_history.sql:162]: It's a matter of choice, let me bring this up internally.

File: proto/minder/v1/minder.proto [hunk 2]
```diff
@@ -3271,6 +3271,23 @@ message ListEvaluationHistoryRequest {
     // Timestamp representing the end time of the selection window.
     google.protobuf.Timestamp to = 9;
 
+    // Filter evaluation history to only those matching the specified labels.
+    //
+    // The default is to return all user-created profiles; the string "*" can
+    // be used to select all profiles, including system profiles.  This syntax
+    // may be expanded in the future.
+    repeated string label_filter = 11 [
```
Comment (eleftherias) [inline proto/minder/v1/minder.proto:3279]: Should this be a single string, similar to what the evaluation history and list profiles requests have?

File: internal/history/models.go [hunk 3]
```diff
@@ -357,6 +374,29 @@ func (filter *listEvaluationFilter) ExcludedProfileNames() []string {
 	return filter.excludedProfileNames
 }
 
+func (filter *listEvaluationFilter) AddLabel(label string) error {
+	if label == "!*" {
```
Comment (eleftherias) [inline internal/history/models.go:378]: This looks like the same logic that exists in `LabelsFromFilter`, should it be centralized somewhere so it's not duplicated? https://github.com/mindersec/minder/blob/main/internal/db/domain.go#L49

File: proto/minder/v1/minder.proto (hunk 2 above)
Comment (blkt) [reply proto/minder/v1/minder.proto:3279]: Filters usable in the history-list endpoint are all repeated string so that they can be sent (e.g. via a REST GET) as separate entries (e.g. `?label_filter=foo&label_filter=bar`), as comma separated strings (e.g. `?label_filter=foo,bar`), or as a combination of the two.
The handler is responsible for splitting the string and populating the struct input to the service/db layer as an array of strings. Layers outside of the control plane only reason in terms of array of strings.

My understanding of `label_filter` in Profiles is that it can be only passed as [comma separated string](https://github.com/mindersec/minder/blob/main/internal/db/domain.go#L58), which is supported in this case as well. Additionally, the database layer is responsible for splitting the string, while here the responsibility is separate, which hopefully makes lower interfaces clearer.

TL;DR: it already support comma separated strings of labels, I would keep it this way.

File: internal/history/models.go (hunk 3 above)
Comment (blkt) [reply internal/history/models.go:378]: I agree that this logic should be shared, but history list RPC splits the responsibility between the controlplane and the service layer, while `LabelsFromFilter` wraps it in the database layer.

I would update how the `LabelsFromFilter` works by doing the same as done here, but in a separate PR. I can open a task for proper prioritization.

**Review by eleftherias — APPROVED** [review]

**Review by eleftherias — APPROVED** [review]
